packages/next/src/build/flying-shuttle/inline-static-env.ts

```diff
@@ -0,0 +1,53 @@
+import fs from 'fs'
+import path from 'path'
+import { promisify } from 'util'
+import globOriginal from 'next/dist/compiled/glob'
+import { getNextPublicEnvironmentVariables } from '../webpack/plugins/define-env-plugin'
+import { Sema } from 'next/dist/compiled/async-sema'
+
+const glob = promisify(globOriginal)
+
+export async function inlineStaticEnv({ distDir }: { distDir: string }) {
+  const staticEnv = getNextPublicEnvironmentVariables()
+
+  const serverDir = path.join(distDir, 'server')
+  const serverChunks = await glob('**/*.js', {
+    cwd: serverDir,
+  })
+  const clientDir = path.join(distDir, 'static')
+  const clientChunks = await glob('**/*.js', {
+    cwd: clientDir,
+  })
+
+  const inlineSema = new Sema(8)
+
+  for (const [parentDir, files] of [
+    [serverDir, serverChunks],
+    [clientDir, clientChunks],
+  ] as const) {
+    await Promise.all(
+      files.map(async (file) => {
+        await inlineSema.acquire()
+        const filepath = path.join(parentDir, file)
+        const content = await fs.promises.readFile(filepath, 'utf8')
+        // TODO: should we support process.env['NEXT_PUBLIC_KEY'] format?
```

Comment:
This looks valid, maybe is necessary in the following PR.

Reply:
I think this one isn't currently supported with DefineEnv so might not need handling but left note to check in followup yeah